docs/specifications/tests/Zone-TP/zone01.md

```diff
@@ -3,43 +3,252 @@
+* [Scope](#scope)
+* [Inputs](#inputs)
+* [Summary](#summary)
+* [Test procedure](#test-procedure)
+* [Outcome(s)](#outcomes)
+* [Special procedural requirements](#special-procedural-requirements)
+* [Intercase dependencies](#intercase-dependencies)
+* [Terminology](#terminology)
 
-### Inputs
+## Objective
 
-The domain name to be tested.
+The MNAME field from the SOA record of a zone is supposed to contain the master
+name server for that zone. The hostname of the MNAME field may not be listed
+among the delegated name servers, but should still be authoritative for the zone.
+MNAME may be used for other services such as DNS NOTIFY described in [RFC1996].
 
-### Ordered description of steps to be taken to execute the test case
+[RFC1035], section 3.3.13, specifies that "the *domain-name* of the name server
+that was the original or primary source of data for this zone".
 
-1. Retrieve the SOA record from a delegated name server for the domain.
-2. If the answer from step 1 is not authoritative, iterate step 1 until there is an authoritative answer.
-3. If the SOA MNAME field does not exist in DNS, this text case fails.
-4. If the IP addresses for the SOA MNAME field does not answer DNS queries
-   this test case fails.
-5. If the SOA MNAME field is not authoritative for the zone tested,
-   this test case fails.
-6. If the SOA MNAME field is not part of the NS set for the zone, this
-   test case may yield a notice message.
+[RFC1996], section 2, and [RFC2136], section 1, add that "the primary master is
+named in the zone's SOA MNAME field and optionally by an NS RR.
+There is by definition only one primary master server per zone".
 
-### Outcome(s)
+[RFC2181], section 7.2, clarifies that "it is quite clear in the specifications,
+yet seems to have been widely ignored, that the MNAME field of the SOA record
+should contain the name of the primary (master) server for the zone identified
+by the SOA. It should not contain the name of the zone itself. That information
+would be useless, as to discover it, one needs to start with the domain name of
+the SOA record - that is the name of the zone".
 
-If the SOA MNAME field is not an authoritative DNS server for the domain name,
-this test case fails.
+There exists an unstandardized practice to set the SOA MNAME to "." to mean no
+server at all, indicating that there is no default server for dynamic updates.
+There is at least one old and expired Internet-Draft that attempted to standardize
+that behavior, i.e. [draft-jabley-dnsop-missing-mname]. If the SOA MNAME
+is an empty name (".") this Test Case will not try to connect to a server behind
+it since there will never be a server behind that name, as the purpose is most
+definitely to follow that practice. Instead, a special message will be outputted.
+ 
+This Test Case will check that:
+ - the SOA MNAME contains the master name server of *Child Zone*,
+   as best as it can be determined.
+ - the master name server is authoritative of *Child Zone*.
```

Comment:
It should rather be
```
- the SOA MNAME name server is authoritative of *Child Zone*.
```

Reply:
Updated

Reply:
I suggested "SOA MNAME name server", not "master name server".